VMware.PSDesiredStateConfiguration.md

```diff
@@ -0,0 +1,147 @@
+For **PowerShell 5.1** you need to enable Windows Remote Management
+For information on how to enable it read here: [WinRM guide](https://docs.microsoft.com/en-us/windows/win32/winrm/installation-and-configuration-for-windows-remote-management)
+
+
+## Installations
+
+### Manual Installation
+
+1. Copy the VMware.PSDesiredStateConfiguration Module to one of the system PowerShell module directories. For more information on installing PowerShell Modules, please visit [Installing a PowerShell Module](https://docs.microsoft.com/en-us/powershell/scripting/developer/module/installing-a-powershell-module?view=powershell-7).
+2. In PowerShell import the VMware.PSDesiredStateConfiguration Module:
+   ```
+    Import-Module -Name 'VMware.PSDesiredStateConfiguration'
+   ```
+
+   To check if the module was successfully installed:
+   ```
+    Get-Module -Name 'VMware.PSDesiredStateConfiguration'
+   ```
+
+## Use Overview
+VMware.PSDesiredStateConfiguration works with ps1 configuration files and the general flow is as follow:
+- Compile ps1 DSC configuration and transforms given configuration to a configuration object. The object contains the name of the configuration and an array of sorted DSC Resource by dependencies. Each Resource contains **InstanceName** for name of the resource given in the configuration, **ResourceType** which display the type of resource,  **ModuleName** which displays the module of the resource, **Property** which is a hashtable of properties which are unique to the given resource. Example Configuration object from a given configuration:
+    ```
+    Configuration Test
+    {
+        Import-DscResource -ModuleName MyDscResource
+
+        CustomResource myResource
+        {
+            Field = "Test field"
+            Ensure = "Present"
+        }
+    }
+
+    # configuration object result
+    [VmwDscConfiguration]@{
+        InstanceName = 'Test'
+        Resource = @(
+            [VmwDscResource]@{
+                InstanceName = 'myResource'
+                ResourceType = 'CustomResource'
+                ModuleName = 'MyDscResource'
+                Property = @{
+                    Field = "Test field"
+                    Ensure = "Present"
+                }
+            }
+        )
+    }
+    ```
+    Note: this step does not require the use of **LCM** and does not generate a **MOF** file.
+- The resulting configuration object generated by the **New-VmwDscConfiguration** cmdlet gets used with the Start, Get, Test cmdlets and the result is based on the cmdlet.
+
+## Cmdlet Guide
+#### New-VmwDscConfiguration
+This cmdlet creates a VmwDscConfiguration object which contains information about the configuration. This object is then used with the other cmdlets in order to perform the three main DSC operations(GET, SET, TEST).
+
+##### Parameters
+- **ConfigName** - Name of the Configuration to be compied as string
+- **CustomParams** - Represents a hashtable of parameters that are used for the Configuration.
```

Comment:
Can you provide examples for the different parameters for a better understanding ?